src/shims/unix/linux/epoll.rs

```diff
@@ -22,13 +34,30 @@ struct Epoll {
 ///
 /// <https://man7.org/linux/man-pages/man2/epoll_ctl.2.html>
 #[derive(Clone, Debug)]
-struct EpollEvent {
+pub struct EpollEvent {
+    #[allow(dead_code)]
+    pub file_descriptor: i32,
+    #[allow(dead_code)]
+    pub weak_file_descriptor: WeakFileDescriptor,
     #[allow(dead_code)]
-    events: u32,
+    pub events: u32,
     /// `Scalar` is used to represent the
     /// `epoll_data` type union.
     #[allow(dead_code)]
-    data: Scalar,
+    pub data: Scalar,
+    #[allow(dead_code)]
+    pub ready_list: Rc<RefCell<BTreeMap<(WeakFileDescriptor, i32), EpollReturn>>>,
+}
+
+// This trait is for any file description that can be monitored by epoll.
+pub trait EpollTarget {
```

Comment:
I think you can just add these methods to `FileDescription` and make them return an InterpResult. This way all the impls that don't care about it can error out with an unsupported error

Reply:
Thanks, this works. Now ``check_readiness`` can also have direct access to the libc flags.